### DISCUSSION TITLE
How to use LongPress behavior on CollectionView with DataTemplate?

### DISCUSSION THREAD

**Discussion by QuadMan** [CommunityToolkit/Maui#1952] [Q&A]
I need an example of the usage of LongPress behavior on CollectionView list.
I use DataTemplate like this:
`<ContentPage x:Name="This">
...
 <CollectionView SelectionChanged="OnCollectionViewSelectionChanged"
                 SelectionMode="{Binding ListSelectionMode}"
                 ItemsSource="{Binding LessonsList}">

     <CollectionView.ItemTemplate>
         <DataTemplate x:DataType="{x:Type model:Lesson}">
             <Border WidthRequest="120" HeightRequest="80">
                 <Border.Behaviors>
                     <toolkit:TouchBehavior                                              
                         LongPressCommandParameter="{Binding .}"
                         LongPressCommand="{Binding Source={x:Reference This}, Path=_vm.LongPressLessonCommand}" 
                         LongPressDuration="1000" />
                 </Border.Behaviors>
				 <Label Text="Test"/>
             </Border>
         </DataTemplate>
     </CollectionView.ItemTemplate>
 </CollectionView>
...
</ContentPage>`

But nothing happens.
Does it should work? Or current version 9.0.1 is not compatible with LongPress behavior in CollectionView DataTemplate?

**Comment by GeorgeLeithead** [comment]
There is a new article created by [Leomaris Reyes](https://www.telerik.com/blogs/author/leomaris-reyes) that goes into using the `LongPress` : [Exploring Long Press in .NET MAUI Community Toolkit](https://www.telerik.com/blogs/exploring-long-press-net-maui-community-toolkit).  This may help.
